components/command/command-handler/src/test/java/uk/gov/justice/services/core/aggregate/AggregateServiceIT.java

```diff
@@ -0,0 +1,378 @@
+import org.apache.openejb.testing.Module;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(ApplicationComposer.class)
+public class AggregateServiceIT {
+
+    private static final UUID STREAM_ID = randomUUID();
+
+    private static final String LIQUIBASE_SNAPSHOT_CHANGELOG_XML = "liquibase/snapshot-store-db-changelog.xml";
+
+    private static final String LIQUIBASE_EVENT_STORE_CHANGELOG_XML = "liquibase/event-store-db-changelog.xml";
+
+    private static final String AGGREGATE_INTERFACE_FULL_NAME = "uk.gov.justice.domain.aggregate.Aggregate";
+
+    private static final String TEST_AGGREGATE_CLASS_NAME = "GeneratedTestAggregate";
+
+    private static final String TEST_AGGREGATE_PACKAGE = "uk.gov.justice.services.core.aggregate";
+
+    private static final String TEST_AGGREGATE_FULL_NAME = format("%s.%s", TEST_AGGREGATE_PACKAGE, TEST_AGGREGATE_CLASS_NAME);
+
+    private static final String TEST_AGGREGATE_COMPILED_CLASS = format("%s/%s.class", TEST_AGGREGATE_PACKAGE.replace(".", "/"), TEST_AGGREGATE_CLASS_NAME);
+    private static final int SNAPSHOT_THRESHOLD = 25;
+
+
+    @Resource(name = "openejb/Resource/eventStore")
+    private DataSource dataSource;
+
+    @Inject
+    private SnapshotOpenEjbAwareJdbcRepository snapshotRepository;
+
+    @Inject
+    private EventSource eventSource;
+
+    @Inject
+    private AggregateService aggregateService;
+
+
+    @Module
+    @Classes(cdi = true, value = {
+            CustomClassLoaderObjectInputStreamStrategy.class,
+            AggregateService.class,
+            SnapshotOpenEjbAwareJdbcRepository.class,
+            EventLogOpenEjbAwareJdbcRepository.class,
+            EventSource.class,
+            EnvelopeEventStream.class,
+            EventStreamManager.class,
+            LoggerProducer.class,
+            EventRepository.class,
+            EventLogConverter.class,
+            StringToJsonObjectConverter.class,
+            JsonObjectEnvelopeConverter.class,
+            JsonObjectToObjectConverter.class,
+            ObjectMapperProducer.class,
+            JdbcEventRepository.class,
+            JmsEventPublisher.class,
+            DummyJmsEnvelopeSender.class,
+            DefaultEventDestinationResolver.class,
+            EnvelopeConverter.class,
+            DefaultSnapshotService.class,
+            DefaultSnapshotStrategy.class
+    })
+
+    public WebApp war() {
+        return new WebApp()
+                .contextRoot("snapshot-test")
+                .addServlet("SnapShotApp", Application.class.getName());
+    }
+
+    @Before
+    public void init() throws Exception {
+        initDatabase();
+        aggregateService.register(new EventFoundEvent(EventA.class, "context.eventA"));
+    }
+
+    @Test
+    public void shouldStoreABrandNewSnapshotWhenEventCountInTheStreamReachesThreshold() throws Exception {
+
+
+        EventStream stream = eventSource.getStreamById(STREAM_ID);
```

Comment:
Should be consistent, some variables are marked as final others are not.  Preference would be all marked as final.